File: OOP/Proyects/De C++ a Java/Herencia/Ejercicio2/Java/src/animales/Pug.java

package animales;

public class Pug extends Perros{
    private String nameOwner;
    
    public Pug(String name, int age, float weight){
        super(name, age, weight);
        nameOwner="Unknown";
    }
    
    public void setNameOwner(String nameOwner){
        this.nameOwner = nameOwner;
    }
    
    public String getNameOwner(){
        return nameOwner;
    }
    
    public void print(){
        System.out.println("Dog's name: "+getName()+" with weight "+getWeight()+". Owner's name: "+getNameOwner());
    }
}

File: OOP/Proyects/De C++ a Java/Herencia/Ejercicio2/C++/Pug.cpp
#include "Pug.h"//Se incluye la clase Pug

Perros::Pug(char, int , float)//Constructor que recibe parametros de tipo char, int y float
{
	typedef(name, age, weiht);
	nameOwner = "Unknow";
}

void Pug::setNameOwner(){//Funcion setNameOwner de tipo void
	this->nameOwner = nameOwner;//this llama la funcion privada nameOwner 
}

char Pug::getNameOwner(){//Funcion getNameOwner de tipo char
	return nameOwner;//Retorna la variable nameOwner
}

void print(){//funcion print de tipo void
	printf("Dog's name: ", getName(), " with weight ", getWeight(),", Owner's name: ", getNameOwner());//Muestra las funciones
}
